### ISSUE TITLE
Build failing on JDK 9 / vavr 0.9.1

### ISSUE THREAD

**Issue by sskorol** [vavr-io/vavr#2097]
I'm using **vavr** in the following project: https://github.com/sskorol/test-data-supplier/tree/35-java-9-support

And it works fine after migration to Java 9 with vavr 0.9.0 version.

Today I've tried to update vavr to 0.9.1, and got the following exception in console log:

```plain
Error occurred during initialization of boot layer
java.lang.module.FindException: Unable to derive module descriptor for /Users/skorol/.gradle/caches/modules-2/files-2.1/io.vavr/vavr/0.9.1/d6707c671271472b397724a0bfafad10239dd260/vavr-0.9.1.jar
Caused by: java.lang.module.InvalidModuleDescriptorException: Provider class io.vavr.match.PatternsProcessor not in module
Process 'Gradle Test Executor 4' finished with non-zero exit value 1
org.gradle.process.internal.ExecException: Process 'Gradle Test Executor 4' finished with non-zero exit value 1
	at org.gradle.process.internal.DefaultExecHandle$ExecResultImpl.assertNormalExitValue(DefaultExecHandle.java:380)
	at org.gradle.process.internal.worker.DefaultWorkerProcess.onProcessStop(DefaultWorkerProcess.java:118)
	at org.gradle.process.internal.worker.DefaultWorkerProcess.access$000(DefaultWorkerProcess.java:41)
	at org.gradle.process.internal.worker.DefaultWorkerProcess$1.executionFinished(DefaultWorkerProcess.java:74)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:564)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:230)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:149)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:140)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:37)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy78.executionFinished(Unknown Source)
	at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:213)
	at org.gradle.process.internal.DefaultExecHandle.finished(DefaultExecHandle.java:325)
	at org.gradle.process.internal.ExecHandleRunner.completed(ExecHandleRunner.java:103)
	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:83)
	at org.gradle.internal.operations.BuildOperationIdentifierPreservingRunnable.run(BuildOperationIdentifierPreservingRunnable.java:39)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:844)
```

Seems like the root cause is that `vavr` module depends on `vavr-match`. And `vavr-match` is not a Java 9 module yet. So automatic naming assumes its jar name as a base module name.

However, `varv-match` is not a valid module name, as we can't use hyphen in `module-info.java`:

```java
requires vavr-match;
```

**Comment by danieldietrich** [comment]
This will be addressed in the upcoming bugfix release 0.9.2.
We work on it in [Initial Spike to support JDK 9](https://github.com/vavr-io/vavr/pull/1994).

**Comment by danieldietrich** [comment]
I think this might be already fixed with #2102.

**Comment by danieldietrich** [comment]
Yes, it is already fixed with #2102. Will release it tomorrow as part of Vavr 0.9.2
